docs/source/chaincode4ade.rst

```diff
@@ -18,58 +18,35 @@ only read query is allowed. That is, the called chaincode on a different channel
-using the Fabric contract API, visit the :doc:`developapps/smartcontract` topic.
+This tutorial provides an overview of the high level APIs provided by the Fabric Contract API.
+To learn more about developing smart contracts using the Fabric contract API, visit the :doc:`developapps/smartcontract` topic.
 
-Chaincode API
--------------
+Fabric Contract API
+-------------------
 
-Every chaincode program must implement the ``Chaincode`` interface whose methods
-are called in response to received transactions. You can find the reference
-documentation of the Chaincode Shim API for different languages below:
+The ``fabric-contract-api`` provides the contract interface, a high level API for application developers to implement Smart Contracts.
+Within Hyperledger Fabric, Smart Contracts are also known as Chaincode. Working with this API provides a high level entry point to writing business logic.
+Documentation of the Fabric Contract API for different languages can be found at the links below:
 
-  - `Go <https://godoc.org/github.com/hyperledger/fabric-chaincode-go/shim#Chaincode>`__
-  - `Node.js <https://hyperledger.github.io/fabric-chaincode-node/{BRANCH}/api/fabric-shim.ChaincodeInterface.html>`__
-  - `Java <https://hyperledger.github.io/fabric-chaincode-java/{BRANCH}/api/org/hyperledger/fabric/shim/Chaincode.html>`__
-
-In each language, the ``Invoke`` method is called by clients to submit transaction
-proposals. This method allows you to use the chaincode to read and write data on
-the channel ledger.
-
-You also need to include an ``Init`` method in your chaincode that will serve as
-the initialization function. This function is required by the chaincode interface,
-but does not necessarily need to invoked by your applications. You can use the
-Fabric chaincode lifecycle process to specify whether the ``Init`` function must
-be called prior to Invokes. For more information, refer to the initialization
-parameter in the `Approving a chaincode definition <chaincode_lifecycle.html#step-three-approve-a-chaincode-definition-for-your-organization>`__
-step of the Fabric chaincode lifecycle documentation.
-
-The other interface in the chaincode "shim" APIs is the ``ChaincodeStubInterface``:
-
-  - `Go <https://godoc.org/github.com/hyperledger/fabric-chaincode-go/shim#ChaincodeStubInterface>`__
-  - `Node.js <https://hyperledger.github.io/fabric-chaincode-node/{BRANCH}/api/fabric-shim.ChaincodeStub.html>`__
-  - `Java <https://hyperledger.github.io/fabric-chaincode-java/{BRANCH}/api/org/hyperledger/fabric/shim/ChaincodeStub.html>`__
-
-which is used to access and modify the ledger, and to make invocations between
-chaincodes.
+  - `Go <https://godoc.org/github.com/hyperledger/fabric-contract-api-go/contractapi>`__
+  - `Node.js <https://hyperledger.github.io/fabric-chaincode-node/{BRANCH}/api/>`__
+  - `Java <https://hyperledger.github.io/fabric-chaincode-java/{BRANCH}/api/org/hyperledger/fabric/contract/package-summary.html>`__
```

Comment:
The ChaincodeStubInterface links can remain.
I'd suggest add a couple sentences mentioning that when using the contract api, each chaincode function that is called is passed a transaction context "ctx", from which you can get the chaincode stub (GetStub() ), which has functions to access the ledger (e.g. GetState() ) and make requests to update the ledger (e.g. PutState() ).

This will be a nice introductory bridge to GetState() and PutState(), which are used in the following code snippets.